test/cmpd_dset.c

```diff
@@ -700,27 +964,126 @@ test_select_compound(char *fname, hid_t fapl, hid_t in_dxpl, unsigned set_fillva
         s3[i].e    = 8 * i + 7;
     }
 
-    /* Save a copy as the buffer may be clobbered due to H5Pset_modify_write_buf() */
     memcpy(save_s3, s3, sizeof(s3_t) * NX * NY);
 
+    /* Set buffer size accordingly */
     if (set_buf) {
-        ss1 = H5Tget_size(s1_tid);
+        /* ss1 is set already previously */
         ss2 = H5Tget_size(s3_tid);
         ss  = MAX(ss1, ss2) * NX * NY;
 
         if (H5Pset_buffer(dxpl, ss, NULL, NULL) < 0)
             goto error;
     }
 
-    /* Write to the dataset with s3_tid */
-    if (H5Dwrite(did, s3_tid, H5S_ALL, H5S_ALL, dxpl, s3) < 0)
+    /* Read from the dataset with mem type s3_tid */
+    if (H5Dread(did, s3_tid, H5S_ALL, H5S_ALL, dxpl, rbuf3) < 0)
+        goto error;
+
+    if (compare_s1_s3_data(save_s1, rbuf3) < 0)
         goto error;
 
-    if (H5Dread(did, s3_tid, H5S_ALL, H5S_ALL, dxpl, rbuf3) < 0)
+    if (H5Dclose(did) < 0)
+        goto error;
+
+    PASSED();
+
+    printf("    test_select_compound(): verify the CI window failure is fixed");
```

Comment:
"CI window" is not very descriptive, even if corrected to "Windows". Maybe something like "test non-optimized compound conversions with memory type larger than file and a too small type conversion buffer", though that's maybe too long?

Reply:
Changed to 
printf("    test_select_compound(): verify fix for non-optimized compound conversions with memory type larger than file ");